router/schema_validator/boolean_schema_validator.go

```diff
@@ -1,16 +1,14 @@
 package schema_validator
 
-import (
-	"fmt"
-	"reflect"
-)
+import "github.com/getkin/kin-openapi/openapi3"
 
-func (c typeSchemaValidatorContext) validateBooleanSchema() error {
-	if c.schema.Type != booleanSchemaType {
-		return nil
+func (c typeSchemaValidatorContext) validateBooleanSchema() {
+
+	if isBoolType(c.goType) && !isSchemaTypeBooleanOrEmpty(c.schema) {
```

Comment:
extract calls to usBoolType and schemaTypeIsIncompatibleWithType to local variables.